lib/remap-lang/src/lib.rs

```diff
@@ -131,6 +131,34 @@ mod tests {
     use super::*;
     use std::collections::HashMap;
 
+    #[derive(Debug, Clone)]
+    struct RegexPrinter;
```

Comment:
Is the program:

```
/foo/
```

Valid? It'd be unusual, but I'd expect that to simply return the regex as the result of the script.

Asking as, if it is, I'm wondering why we need this printer.

Reply:
No, that's unsupported, regexes are only valid as function arguments (which is indeed why we need this printer function).

See the relevant grammar:

https://github.com/timberio/vector/blob/a87b132e6c00c56467a261a42a52be564962dc54/lib/remap-lang/grammar.pest#L22

Reply:
Aha :+1: . I'm wondering if treating regexes as values would simplify things (or make them more complicated :smile: ).

For example, it would allow things like:

```
$my_regex = /foo/
replace(.message, $my_regex, "something")
```

I suppose an issue is that `event::Value` has no concept of regexes.

Reply:
To expand on that:

> but I'd expect that to simply return the regex as the result of the script

What `Value` type would you expect this to return? Introducing a special `Regex` value type seems unwanted, as we can't serialize that to anything meaningful in JSON for example, unless we wrap the regex in a string (but then we can't distinguish it from a regular string when deserializing that same JSON payload.

Alternatively, we could have it serialize to some kind of map with a `kind` field and a `regex` field, and in fact that's what we initially pursued, but we considered the use-case to be so exotic, meanwhile it could introduce unknown performance bottlenecks since you'd have to parse the regex every time a new event is captures that contains a "regex field".

So in the end, we decided it was best to just limit regexes to function arguments, and hard-code them into the language grammar, instead of representing them as a `Value`.